lua/telescope/actions/init.lua

```diff
@@ -257,6 +265,44 @@ actions.insert_symbol = function(prompt_bufnr)
   vim.api.nvim_put({selection.value[1]}, '', true, true)
 end
 
+actions.create_file = function(prompt_bufnr)
+  local file = vim.fn.input('Create file or directory(add ' ..
+                            path.separator ..
+                            ' at the end of file). Enter name: ')
+  if file == "" then return end
+
+  local fpath = actions.get_selected_entry(prompt_bufnr).path
+  if string.sub(fpath, -1) == path.separator then
+    fpath = vim.fn.fnamemodify(fpath, ':h:h')
+  else
+    fpath = vim.fn.fnamemodify(fpath, ':h')
+  end
+
+  fpath = fpath .. path.separator .. file
+
+  if string.sub(fpath, -1) ~= path.separator then
+    actions.close(prompt_bufnr)
+    path.create_file(fpath)
+    vim.cmd(string.format(':e %s', fpath))
+  else
+    do_close(prompt_bufnr, true)
+    -- TODO(conni2461): Recursive
```

Comment:
`vim.fn.mkdir` I think has an option that will auto create all the required directories

Reply:
Kinda want to do it with plenary.path and uv :) you already wrote the test but its still pending. I just haven't found the motivation to do it